dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/experiment/ExperimentCodeInjecterViewTool.java

```diff
@@ -0,0 +1,40 @@
+package com.dotcms.rendering.velocity.viewtools.experiment;
+
+import com.dotcms.api.web.HttpServletRequestThreadLocal;
+import com.dotcms.experiments.business.ExperimentCodeGenerator;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.liferay.portal.model.User;
+import com.liferay.util.StringPool;
+import javax.servlet.http.HttpServletRequest;
+import org.apache.velocity.tools.view.tools.ViewTool;
+
+/**
+ * ViewTool to inject the JS/HTML Experiment Code
+ */
+public class ExperimentCodeInjecterViewTool implements ViewTool {
+
+    @Override
+    public void init(Object initData) {
+
+    }
+
+    @Override
+    public User getUser(HttpServletRequest request) {
+        return ViewTool.super.getUser(request);
+    }
+
+    public String code(){
+        try {
+            final Host currentHost = WebAPILocator.getHostWebAPI().getCurrentHost();
+            final HttpServletRequest request = HttpServletRequestThreadLocal.INSTANCE.getRequest();
+            return ExperimentCodeGenerator.INSTANCE.getCode(currentHost, request).orElse(StringPool.BLANK);
```

Comment:
May be you can put this into the WebAPILocator

Reply:
done https://github.com/dotCMS/core/pull/24321/commits/3dfbcf6cd202e85cfcdf8cd509c2990d16e05b48#diff-cbb6bb76d043d2cb96b6503fafe6b5b1e5525c9b6fb58844a57a397b1f037cb6R57